### PR TITLE
Allow importing of mixed case nucleotide sequences.

### PR DESCRIPTION
**Pull request by mikerobeson**
This PR allows mixed-case DNA and RNA sequence files to be imported, in an attempt to  address #91. For example, if a user would like to import lowercase DNA sequence files (e.g. [RDP reference sequences](https://rdp.cme.msu.edu/misc/resources.jsp)), they can now do so by running the following import statement with the new format `MixedCaseDNAFASTAFormat`:

```
qiime tools import \
    --input-path current_Archaea_unaligned.fa \
    --output-path current_Archaea_unaligned.qza \
    --input-format MixedCaseDNAFASTAFormat \
    --type 'FeatureData[Sequence]'
```

The associated `formats` and `transformers` were added (_no new `types`_):

- Format to format conversion
  - `MixedCaseDNAFASTAFormat` --> `DNAFASTAFormat`
  - `MixedCaseRNAFASTAFormat` --> `RNAFASTAFormat`
  - `MixedCaseAlignedDNAFASTAFormat` --> `AlignedDNAFASTAFormat`
  - `MixedCaseAlignedRNAFASTAFormat` --> `AlignedRNAFASTAFormat`
- Format to iterator conversion:
  -  `MixedCaseDNAFASTAFormat` --> `DNAIterator`
  - `MixedCaseRNAFASTAFormat` --> `RNAIterator`
  - `MixedCaseAlignedDNAFASTAFormat` --> `AlignedDNAIterator`
  - `MixedCaseAlignedRNAFASTAFormat` --> `AlignedRNAIterator`
- All formats to series.
- All formats to metadata.

Essentially, all of the `MixedCase*` formats are converted to the uppercase IUPAC characters by using the `lowercase=True` option of `skbio.DNA|RNA` constructors. Writing out `MixedCase*` formats is not supported in this PR, as to do so would require making a new `skbio.sequence.GrammaredSequence` class to enable proper iterator, reader, writer functionality for mixed case types. That is `skbio.DNA|RNA` enforce IUPAC characters (*i.e.* capital letters). _I can do this at a later date if others think it is worthwhile, but I do not see an immediate use case._

### CODE REVIEW

**Comment by mikerobeson** [comment]
I am not sure why I am getting this list error:
```
./q2_types/feature_data/__init__.py:11:1: F401 '._format.MixedCaseDNASequencesDirectoryFormat' imported but unused
./q2_types/feature_data/__init__.py:11:1: F401 '._format.MixedCaseRNAFASTAFormat' imported but unused
```
If I remove them I get a "most likely due to a circular import" issue...

Any ideas?

File: q2_types/feature_data/__init__.py [hunk 1]
```diff
@@ -47,6 +53,12 @@
     'RNAFASTAFormat', 'AlignedRNAFASTAFormat', 'RNASequencesDirectoryFormat',
     'AlignedRNASequencesDirectoryFormat', 'RNASequence', 'AlignedRNASequence',
     'PairedRNAIterator', 'PairedRNASequencesDirectoryFormat',
-    'PairedEndRNASequence', 'BLAST6Format', 'BLAST6DirectoryFormat', 'BLAST6']
+    'PairedEndRNASequence', 'BLAST6Format', 'BLAST6DirectoryFormat', 'BLAST6',
+    'MixedCaseDNAFASTAFormat', 'MixedCaseDNASequencesDirectoryFormat'
```
Comment (nbokulich) [inline q2_types/feature_data/__init__.py:57]: you missed the comma at the end of this line — this is causing the misleading "imported but not used" error.

**Comment by Oddant1** [comment]
@mikerobeson @nbokulich is this PR good to merge for the coming release?

**Review by ebolyen — APPROVED** [review]
Wow, there's even tests for everything! This is really great, thank you @mikerobeson!

(And thanks for reviewing earlier @nbokulich!)

**Pull request merged**